doc/sphinx/language/core/records.rst

```diff
@@ -3,208 +3,313 @@
 
-.. table:: Printing Constructor @qualid
+   .. example:: Constructing 1/2 as a record
 
-   This :term:`table` specifies a set of qualids which are displayed as constructors.  Use the
-   :cmd:`Add` and :cmd:`Remove` commands to update the set of qualids.
+      Constructing the rational :math:`1/2` using either the record or application syntax:
 
-This syntax can also be used for pattern matching.
+      .. coqtop:: in
 
-.. coqtop:: all
+         Theorem one_two_irred : forall x y z:nat, x * y = 1 /\ x * z = 2 -> x = 1.
+         Admitted.
 
-   Eval compute in (
-     match half with
-     | {| sign := true; top := n |} => n
-     | _ => 0
-     end).
+         (* Record form: top and bottom can be inferred from other fields *)
+         Definition half :=
+           {| negative := false;
+              Rat_bottom_nonzero := O_S 1;
+              Rat_irreducible := one_two_irred |}.
 
-The :term:`command` generates also, when it is possible, the projection
-functions for destructuring an object of type :token:`ident`. These
-projection functions are given the names of the corresponding
-fields. If a field is named `_` then no projection is built
-for it. In our example:
+         (* Application form: use the constructor and provide values for all the fields
+            in order.  "mkRat" is defined by the Record command *)
+         Definition half' := mkRat true 1 2 (O_S 1) one_two_irred.
 
-.. coqtop:: all
+Accessing fields (projections)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  Eval compute in top half.
-  Eval compute in bottom half.
-  Eval compute in Rat_bottom_cond half.
+   .. insertprodn term_projection term_projection
 
-An alternative syntax for projections based on a dot notation is
-available:
+   .. prodn::
+      term_projection ::= @term0 .( @qualid {? @univ_annot } {* @arg } )
+      | @term0 .( @ @qualid {? @univ_annot } {* @term1 } )
 
-.. coqtop:: all
+   The value of a field can be accessed using *projection form* (:n:`@term_projection`,
+   shown here) or with *application form* (see :n:`@term_application`) using the
+   projection function associated with the field.  Don't forget the parentheses for the
+   projection form.
+   Glossing over some syntactic details, the two forms are:
 
-   Eval compute in half.(top).
+   - :n:`@qualid__record.( {? @ } @qualid__field {* @arg })` (projection) and
 
-.. flag:: Printing Projections
+   - :n:`{? @ } @qualid__field {* @arg } @qualid__record` (application)
 
-   This :term:`flag` activates the dot notation for printing.
+   where the :n:`@arg`\s are the parameters of the inductive type.  If :n:`@` is
+   specified, all implicit arguments must be provided.
 
-   .. example::
+   Since the projected object is part of the notation, it is always
+   considered an explicit argument of :token:`qualid`, even if it is
+   formally declared as implicit (see :ref:`ImplicitArguments`).
+
+   .. exn:: @ident cannot be defined because it is informative and @ident is not
```

Comment:
Clarifies that these are two different idents, also it looks like this is a warning, not an error:

```suggestion
   .. warn:: @ident__1 cannot be defined because it is informative and @ident__2 is not
```

This could also be:
```suggestion
   .. warn:: @ident__proj cannot be defined because it is informative and @ident__record is not
```

BTW, is this the right place to mention this warning, or should it be closer to the definition of record types?